birdsong/views/activate.py

```diff
@@ -0,0 +1,46 @@
+
+from django.conf import settings
+from django.utils.encoding import force_str
+from django.utils.http import urlsafe_base64_decode
+from django.shortcuts import get_object_or_404, render
+from django.http import Http404
+
+from wagtail.models import Site
+
+from birdsong.conf import BIRDSONG_ACTIVATION_TEMPLATE
+from birdsong.models import Contact
+
+
+def activate(request, cidb64, token):
+    """Subscription activation endpoint
+
+    :param request: Request sent to this endpoint
+    :type request: class:`requests.models.Request`
+    :param cidb64: Base64 encoded contact ID (e.g. "YTEwMjA0MzctMGQwYS00ZjBjLWE4MzctMWM2YjgzMDQ3ZjM4")
+    :type cidb64: str
+    :param token: Subscription Activation Token (e.g. "bkwxpq-cc9a685f0e58c20baacf0ce2c93823f3")
+    :type token: str
+
+    :return: Rendered `activate.html` template as an HTTP Response (see `BIRDSONG_ACTIVATION_TEMPLATE`)
+    :rtype: class:`django.http.HttpResponse`
+    """
+    try:
+        cid = force_str(urlsafe_base64_decode(cidb64))
+        contact = get_object_or_404(Contact, pk=cid) # NOTE: It might be more appropriate to use BIRDSONG_CONTACT_CLASS
+    except (TypeError, ValueError, OverflowError, Contact.DoesNotExist):
+        raise Http404 # unable to identify contact, let's pretend this endpoint doesn't exist
+
+    if not contact.check_token(token): # token not valid?
+        raise Http404() # invalid token, let's pretend this endpoint doesn't exist
+
+    if not contact.is_active: # contact not active yet?
+        contact.is_active = True
+        contact.save()
+
+    return render(
```

Comment:
I'm wondering if a redirect makes more sense here? The `activate.html` is pretty minimal, it wont be styled like the site

Reply:
I think a confirmation page upon activation is quite standard and somewhat vital functionality of a subscribe form as it signals success to the end user. Yes, it is not styled as of now but neither is the `unsubscribe.html` page.

At the moment developers can implement their own page by using `BIRDSONG_ACTIVATION_TEMPLATE` setting. This isn't a perfect solution but here is what I think we should do to address the underlying problem:

Moving forward I think we can do better by implementing Birdsong's own base template. That would simplify things and then we would only need to instruct developers to overload that base template (or teach birdsong to integrate with their own), rather than asking them to overload individual html pages.